rules/no-import-test-files.js

```diff
@@ -0,0 +1,102 @@
+'use strict';
+const path = require('path');
+const arrify = require('arrify');
+const pkgUp = require('pkg-up');
+const multimatch = require('multimatch');
+const util = require('../util');
+
+const defaultFiles = [
+	'test.js',
+	'test-*.js',
+	'test/**/*.js',
+	'**/__tests__/**/*.js',
+	'**/*.test.js'
+];
+
+function isTestFile(files, rootDir, sourceFile, importedFile) {
+	const absoluteImportedPath = path.resolve(path.dirname(sourceFile), importedFile);
+	const relativePath = path.relative(rootDir, absoluteImportedPath);
+
+	console.log(sourceFile, importedFile, absoluteImportedPath, relativePath);
+
+	return multimatch([relativePath], files).length === 1;
+}
+
+function getPackageInfo() {
+	const packageFilePath = pkgUp.sync();
+
+	return {
+		rootDir: packageFilePath && path.dirname(packageFilePath),
+		files: util.getAvaConfig(packageFilePath).files
+	};
+}
+
+const create = context => {
+	const filename = context.getFilename();
+
+	if (filename === '<text>') {
+		return {};
+	}
+
+	const packageInfo = getPackageInfo();
```

Comment:
Could you rename this to `projectInfo` / `getProjectInfo()` etc? That better aligns with how things are named in AVA itself.